### ISSUE TITLE
sql: *log.safeError: conn_executor.go:521: panic while executing 1 statements: ROLLBACK TRANSACTION: caused by <redacted>

### ISSUE THREAD

**Issue by tbg** [cockroachdb/cockroach#30777]
https://sentry.io/cockroach-labs/cockroachdb/issues/704004461/

```
*log.safeError: conn_executor.go:521: panic while executing 1 statements: ROLLBACK TRANSACTION: caused by <redacted>
  File "github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go", line 489, in func2
  File "github.com/cockroachdb/cockroach/pkg/sql/txn_state.go", line 231, in finishSQLTxn
  File "github.com/cockroachdb/cockroach/pkg/sql/conn_fsm.go", line 380, in func11
  File "github.com/cockroachdb/cockroach/pkg/util/fsm/fsm.go", line 107, in apply
  File "github.com/cockroachdb/cockroach/pkg/util/fsm/fsm.go", line 136, in ApplyWithPayload
...
(6 additional frame(s) were not displayed)

conn_executor.go:521: panic while executing 1 statements: ROLLBACK TRANSACTION: caused by <redacted>
```

**Comment by jordanlewis** [comment]
```
func (ts *txnState2) finishSQLTxn(connCtx context.Context) {
        ts.mon.Stop(ts.Ctx)
        if ts.cancel != nil {
                ts.cancel()
                ts.cancel = nil
        }
        if ts.sp == nil {
                panic("No span in context? Was resetForNewSQLTxn() called previously?")
        }

        if !ts.mu.txn.IsFinalized() {
                panic(fmt.Sprintf(
                        "attempting to finishSQLTxn(), but KV txn is not finalized: %+v", ts.mu.txn))
        }
```

It's the second panic there. This is from 2.0.3. I think it's probably fixed. @andreimatei?

**Comment by andreimatei** [comment]
I can't find anything on the 2.0 branch that would fix it. And without a repro, it's hard to do anything. Hopefully it is fixed on the 2.1 branch just because client package (which deals with that "finalizing" of a txn) is now supposedly better. Superficially, that assertion no longer exists in that form :P.
Let's leave this open for a while and see if new info shows up on 2.0. Hopefully we can silently close this later.

**Comment by jordanlewis** [comment]
Closing because it's not reproducible, not enough information to go on, and from the 2.0 branch.
